libtenzir/builtins/operators/every_cron.cpp

```diff
@@ -120,19 +133,20 @@ class every_operator final : public operator_base {
       if (done) {
         break;
       }
-      now = time::clock::now();
-      const auto delta = next_run - now;
+      const auto now = time::clock::now();
+      const duration delta = next_run - now;
       if (delta < duration::zero()) {
-        next_run = now + interval;
+        next_run = scheduler_.next_after(now);
         continue;
       }
-      next_run += interval;
+      next_run = scheduler_.next_after(next_run);
+
```

Comment:
You'll get used to the style, but just as a reminder: No blank lines in function bodies.